Sources/SwiftDiagnostics/FixIt.swift

```diff
@@ -36,6 +36,33 @@ public struct FixIt: Sendable {
     case replaceLeadingTrivia(token: TokenSyntax, newTrivia: Trivia)
     /// Replace the trailing trivia on the given token
     case replaceTrailingTrivia(token: TokenSyntax, newTrivia: Trivia)
+    /// Replace a child of a parent node
+    case replaceChild(
+      _ keyPath: AnyKeyPath & Sendable,
+      parent: Syntax,
+      child: Syntax,
+      replacingRange: @Sendable () -> Range<AbsolutePosition>
+    )
```

Comment:
Interesting idea with making `replacingChild` a member of `replaceChild`. I am a little worried that it’s possible to construct a `replaceChild` case with `parent`, `child`, `keyPath` and `replacingRange` members that are incompatible with each other if a user doesn’t use the initializer you defined.

Instead, what do you think about storing the type safe generic version of the data and then expose all the necessary information through a protocol? Something along the lines of the following (really not fleshed out yet, just to give an idea).

```swift
protocol AnyReplacingChildData {
  var replacementRange: Range<AbsoluetePosition>
  var newChild: Syntax
}

struct ReplacingChildData<Parent, Child>: AnyReplacingChildData {
  let parent: Parent
  let newChild: Child
  let keyPath: WritableKeyPath<Parent, Child>
}

case replaceChild(any AnyReplacingChildData)
```